stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/experimental/plugin/logic/LogicRegistry.kt

```diff
@@ -100,4 +100,16 @@ internal class LogicRegistry internal constructor(private val stateRegistry: Sta
         channels.clear()
         threads.clear()
     }
+
+    internal companion object {
+        private var instance: LogicRegistry? = null
+
+        internal fun getOrCreate(stateRegistry: StateRegistry): LogicRegistry {
+            return instance ?: LogicRegistry(stateRegistry).also { logicRegistry ->
+                instance = logicRegistry
+            }
+        }
+
+        internal fun get(): LogicRegistry? = instance
```

Comment:
Do we need this or we can only keep `getOrCreate` instead?

Reply:
👍 we can probably move `requireNotNull` check here and return the non-nullable `LogicRegistry` instead. If we do so, we should also mark this method with `@Throws` annotation